lib/pos_blockchain.go

```diff
@@ -100,160 +101,165 @@ func (bc *Blockchain) processBlockPoS(block *MsgDeSoBlock, currentView uint64, v
 		}
-		return blockNode, errors.Wrap(err, "processBlockPoS: Problem initializing UtxoView: ")
+		return blockNode, nil
 	}
-	// Connect this block to the UtxoView.
+
+	// Connect this block to the parent block's UtxoView.
 	txHashes := collections.Transform(block.Txns, func(txn *MsgDeSoTxn) *BlockHash {
 		return txn.Hash()
 	})
-	if _, err = utxoView.ConnectBlock(block, txHashes, true, nil, block.Header.Height); err != nil {
-		// If it doesn't connect, we want to mark it as ValidateFailed.
-		blockNode, storeInBlockIndexErr = bc.storeValidateFailedBlockInBlockIndex(block)
-		if storeInBlockIndexErr != nil {
-			return blockNode, errors.Wrapf(err, "isValidPoSBlock: Problem adding validate failed block to block index: %v", storeInBlockIndexErr)
-		}
-		return blockNode, errors.Wrap(err, "isValidPoSBlock: Problem connecting block to UtxoView: ")
-	}
 
-	validatorsByStake, err := utxoView.GetAllSnapshotValidatorSetEntriesByStake()
-	if err != nil {
-		// Here's another case where we want to store the block even if we hit an error.
-		// We're not SURE if it's invalid, so we should store it, but not mark is as ValidateFailed
-		blockNode, storeInBlockIndexErr = bc.storeBlockInBlockIndex(block)
-		if storeInBlockIndexErr != nil {
-			return blockNode, errors.Wrapf(err, "isValidPoSBlock: Problem adding block to block index: %v", storeInBlockIndexErr)
-		}
-		return blockNode, errors.Wrap(err, "isValidPoSBlock: Problem getting validator set: ")
-	}
-	// TODO(diamondhands): I don't think the TODO below makes sense given the changes but take a look.
-	// TODO: If block belongs in the next epoch or it's from an epoch far ahead in the future, we may not be able to validate its QC at all.
-	// the validator set for that epoch may be entirely different.
+	// TODO: If block belongs in the next epoch or it's from an epoch far ahead in the future, we may not be able to
+	// validate its QC at all. the validator set for that epoch may be entirely different.
 	// A couple of options on how to handle:
 	//   - Add utility to UtxoView to fetch the validator set given an arbitrary block height. If we can't fetch the
 	//     validator set for the block, then we reject it (even if it later turns out to be a valid block)
 	//   - Add block to the block index before QC validation such that even if we aren't able to fetch the validator
 	//     set for the block, we can at least store it locally.
 	// 2. Validate QC
+	validatorsByStake, err := utxoView.GetAllSnapshotValidatorSetEntriesByStake()
+	if err != nil {
+		// Here we want to at least store this block since we don't know if it's invalid.
+		// Hitting an error here should really never happen.
+		blockNode, innerErr := bc.storeBlockInBlockIndex(block)
+		if innerErr != nil {
+			return nil, errors.Wrapf(err, "validateAndIndexBlockPoS: Problem adding block to block index: %v", innerErr)
+		}
+		return blockNode, nil
```

Comment:
We'll be able to get rid of this since we go ahead and store the block on line 153
```suggestion
		return blockNode, nil
```